contracts/programs/gumball-machine/src/lib.rs

```diff
@@ -264,8 +278,17 @@ fn find_and_mint_compressed_nfts<'info>(
     let (mut header_bytes, config_data) =
         gumball_machine_data.split_at_mut(std::mem::size_of::<GumballMachineHeader>());
     let gumball_header = GumballMachineHeader::load_mut_bytes(&mut header_bytes)?;
+
+    // Cannot dispense before all indices are initialized
+    assert_eq!(gumball_header.max_items, gumball_header.smallest_uninitialized_index);
+
+    // Cannot dispense more than the max_mint_size
+    assert!(num_items <= gumball_header.max_mint_size);
```

Comment:
@jarry-xiao I was sort of guessing what this was supposed to be used for lol. Is this what you had in mind?

Reply:
yeah I think this looks reasonable